examples/two-columns-layout/src/plugins/popularCategoriesPlugin.tsx

```diff
@@ -0,0 +1,92 @@
+/** @jsx h */
+import {
+  AutocompleteComponents,
+  AutocompletePlugin,
+  getAlgoliaFacets,
+} from '@algolia/autocomplete-js';
+import { h } from 'preact';
+
+import { ALGOLIA_PRODUCTS_INDEX_NAME } from '../constants';
+import { searchClient } from '../searchClient';
+import { PopularCategoryHit } from '../types';
+
+const images = {
+  women:
+    'https://res.cloudinary.com/hilnmyskv/image/upload/v1646067858/women_category_vwzkln.jpg',
+  bags:
+    'https://res.cloudinary.com/hilnmyskv/image/upload/v1646067858/bags_category_qd7ssj.jpg',
+  clothing:
+    'https://res.cloudinary.com/hilnmyskv/image/upload/v1646067858/clothing_category_xhiz1s.jpg',
+  men:
+    'https://res.cloudinary.com/hilnmyskv/image/upload/v1646067858/men_category_wfcley.jpg',
+  't-shirts':
+    'https://res.cloudinary.com/hilnmyskv/image/upload/v1646067858/t-shirts_category_gzqcvd.jpg',
+  shoes:
+    'https://res.cloudinary.com/hilnmyskv/image/upload/v1646068349/shoes_category_u4fi0q.jpg',
+};
+
+export const popularCategoriesPlugin: AutocompletePlugin<
+  PopularCategoryHit,
+  {}
+> = {
+  getSources() {
+    return [
+      {
+        sourceId: 'popularCategoriesPlugin',
+        getItems() {
+          return getAlgoliaFacets<PopularCategoryHit>({
+            searchClient,
+            queries: [
+              {
+                indexName: ALGOLIA_PRODUCTS_INDEX_NAME,
+                facet: 'list_categories',
+                params: {
+                  facetQuery: '',
+                  maxFacetHits: 6,
+                },
+              },
+            ],
+          });
+        },
+        getItemInputValue({ item }) {
+          return item.label;
+        },
+        onSelect({ setIsOpen }) {
+          setIsOpen(true);
+        },
+        templates: {
+          header() {
+            return (
+              <div className="aa-SourceHeaderTitle">Popular categories</div>
+            );
+          },
+          item({ item, components }) {
+            return <CategoryItem hit={item} components={components} />;
+          },
+        },
+      },
+    ];
+  },
+};
+
+type CategoryItemProps = {
+  hit: PopularCategoryHit;
+  components: AutocompleteComponents;
+};
+
+const CategoryItem = ({ hit }: CategoryItemProps) => {
+  return (
+    <div className="aa-ItemWrapper aa-PopularCategoryItem">
+      <div className="aa-ItemContent">
+        <div className="aa-ItemPicture">
+          <img src={images[hit.label.toLowerCase()]} alt={hit.label} />
```

Comment:
is the label in the hit in inconsistent casing? Would be nice if we could use the label directly

Reply:
We could use it as it is (which the first letter in capital)

Reply:
this way it has the advantage that if we index something with the wrong casing, it will still show up. Will this index get modified ever?

Reply:
I'm using the same one as the UI Template, but it has been duplicated for this example so it shouldn't be modified

Reply:
Should we keep this as it is or remove the `toLowerCase` in your opinion?

Reply:
I think the exact casing is better, but I don't have a strong opinion